libraries/chain/include/graphene/chain/protocol/asset_ops.hpp

```diff
@@ -442,11 +442,42 @@ namespace graphene { namespace chain {
       void            validate()const;
    };
 
+   /**
+    * @brief Transfers BTS from the fee pool of a specified asset back to the issuer's balance
+    
+    * @param fee Payment for the operation execution
+    * @param issuer Account which will be used for transfering BTS
+    * @param asset_id Id of the asset whose fee pool is going to be drained
+    * @param amount_to_claim Amount of BTS to claim from the fee pool
+    * @param extensions Field for future expansion
+    
+    * @pre @ref fee must be paid in the asset other than the one whose pool is being drained
+    * @pre @ref amount_to_claim should be specified in the core asset
+    * @pre @ref amount_to_claim should be nonnegative
+    */
+   struct asset_claim_pool_operation : public base_operation
+   {
+      struct fee_parameters_type {
+         uint64_t fee = 20 * GRAPHENE_BLOCKCHAIN_PRECISION;
+      };
+        
+      asset           fee;
+      account_id_type issuer;
+      asset_id_type   asset_id;        /// fee.asset_id must != asset_id
+      asset           amount_to_claim; /// core asset
```

Comment:
It would be sufficient to specify the amount only, since asset_id is fixed anyway.

Reply:
Correct ... asset contains asset_id ..
Why not change asset to share_type then?

Reply:
How would the blockchain know which asset I want to drain the fee pool of?
We can discuss whether amount_to_claim should be share_type because it will always be CORE/1.3.0 ... but to identify the fee pool, i need the asset_id, no?

Reply:
I meant the amount_to_claim.asset_id - that's always the core asset id.